src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/DotNetCoreProjectCompatibilityDetector.cs

```diff
@@ -254,28 +286,150 @@ public int OnAfterBackgroundSolutionLoadComplete()
+                }
+
+                return CompatibilityLevel.NotSupported;
             }
-            else if(version.Major > s_partialSupportedVersion.Major || (version.Major == s_partialSupportedVersion.Major && version.Minor > s_partialSupportedVersion.Minor))
+
+            // Only has an unsupported version
+            if (compatData.UnsupportedVersion != null)
             {
+                if (version < compatData.UnsupportedVersion)
+                {
+                    return CompatibilityLevel.Recommended;
+                }
+
                 return CompatibilityLevel.NotSupported;
             }
 
-            return CompatibilityLevel.Supported;
+            // No restrictions
+            return CompatibilityLevel.Recommended;
+        }
+
+        /// <summary>
+        /// Pings the server to download version compatibility information and stores this in a cached file in the users app data. If the cached file is
+        /// less than 24 hours old, it uses that data. Otherwise it downloads from the server. If the download fails it will use the previously cached
+        ///  file, or if that file doesn't not exist, it uses the data baked into this class
+        /// </summary>
+        private async Task<VersionCompatibilityData> GetVersionCmpatibilityDataAsync()
+        {
+            try
+            {
+                // Do we meed to update our cached data?
+                if (_curVersionCompatibilityData == null || _timeCurVersionDataLastUpdatedUtc.AddHours(CacheFileValidHours).IsLaterThan(DateTime.UtcNow))
+                {
+                    string downLoadedVersionData = null;
+
+                    // First try the cache
+                    Dictionary<Version, VersionCompatibilityData> versionCompatData = GetCompabilityDataFromCacheFile(checkTimeStamp: true);
+
+                    if (versionCompatData == null)
+                    {
+                        // Try downloading it 
+                        try
+                        {
+                            var httpClient = new DefaultHttpClient();
+                            downLoadedVersionData = await httpClient.GetStringAsync(new Uri(VersionCompatibilityFwlink)).ConfigureAwait(false);
+                            versionCompatData = VersionCompatibilityData.DeserializeVersionData(downLoadedVersionData);
+                        }
+                        catch
+                        {
+                            // If we have a cached file use it
+                            versionCompatData = GetCompabilityDataFromCacheFile(checkTimeStamp: false);
+                        }
+                    }
+
+                    if (versionCompatData != null)
+                    {
+                        // Cache the data locally
+                        if (downLoadedVersionData != null)
+                        {
+                            FileSystem.Value.WriteAllText(VersionDataCacheFile, downLoadedVersionData);
```

Comment:
`VersionDataCacheFile` is only set if `GetLocalAppDataFolderAsync` works.  Do we need to handle the case where it doesn't?

Reply:
If that doesn't work there are serious problems already in VS, but I'll add a null check.